<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson" />


<title>Post-Alignment QC</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.9.3/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PDX_Gar15-13_Veh_DHT_Sarm_RNASeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="description.html">Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="qc_raw.html">Raw Data</a>
    </li>
    <li>
      <a href="qc_trimmed.html">Trimmed Data</a>
    </li>
    <li>
      <a href="qc_aligned.html">Aligned Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge_analysis.html">Differential Gene Expression</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Post-Alignment QC</h1>
<h4 class="author">Stephen Pederson</h4>
<h4 class="date">14 January, 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200930code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200930)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200930code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200930)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLPDXGar1513VehDHTSarmRNASeqtreeb96cfe6b5f4ed09e854fd8ff70de6bb6d73de0f3targetblankb96cfe6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/tree/b96cfe6b5f4ed09e854fd8ff70de6bb6d73de0f3" target="_blank">b96cfe6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLPDXGar1513VehDHTSarmRNASeqtreeb96cfe6b5f4ed09e854fd8ff70de6bb6d73de0f3targetblankb96cfe6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/tree/b96cfe6b5f4ed09e854fd8ff70de6bb6d73de0f3" target="_blank">b96cfe6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    PDX_Gar15-13_Veh_DHT_Sarm_RNASeq.Rproj

Unstaged changes:
    Modified:   analysis/dge_analysis.Rmd
    Modified:   config/config.yml
    Modified:   config/samples.tsv
    Modified:   output/genesGR.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/qc_aligned.Rmd</code>) and HTML (<code>docs/qc_aligned.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/blob/b96cfe6b5f4ed09e854fd8ff70de6bb6d73de0f3/analysis/qc_aligned.Rmd" target="_blank">b96cfe6</a>
</td>
<td>
Steve Ped
</td>
<td>
2021-01-14
</td>
<td>
Recompiled QC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/4ab5279ba0abcba184ad82430844d437f24445ed/docs/qc_aligned.html" target="_blank">4ab5279</a>
</td>
<td>
Steve Pederson
</td>
<td>
2021-01-13
</td>
<td>
Run completed. Needs checking
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/blob/13981da4e4aa349805adf1d9e18b282c16a24974/analysis/qc_aligned.Rmd" target="_blank">13981da</a>
</td>
<td>
Steve Pederson
</td>
<td>
2021-01-13
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ngsReports)
library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(plotly)
library(edgeR)
library(ggfortify)
library(AnnotationHub)
library(ensembldb)
library(magrittr)</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>config &lt;- here::here(&quot;config/config.yml&quot;) %&gt;%
  read_yaml()
suffix &lt;- paste0(config$tag)
sp &lt;- config$ref$species %&gt;%
  str_replace(&quot;(^[a-z])[a-z]*_([a-z]+)&quot;, &quot;\\1\\2&quot;) %&gt;%
  str_to_title()</code></pre>
<pre class="r"><code>samples &lt;- config$samples %&gt;%
  here::here() %&gt;%
  read_tsv() %&gt;%
  mutate(
    Filename = paste0(sample, suffix)
  )</code></pre>
<pre class="r"><code>config$analysis &lt;- config$analysis %&gt;%
  lapply(intersect, y = colnames(samples)) %&gt;%
  .[vapply(., length, integer(1)) &gt; 0]</code></pre>
<pre class="r"><code>if (length(config$analysis)) {
  samples &lt;- samples %&gt;%
    unite(
      col = group, 
      any_of(as.character(unlist(config$analysis))), 
      sep = &quot;_&quot;, remove = FALSE
    )
} else {
  samples$group &lt;- samples$Filename
}</code></pre>
<pre class="r"><code>group_cols &lt;- hcl.colors(
  n = length(unique(samples$group)), 
  palette = &quot;Zissou 1&quot;
  ) %&gt;%
  setNames(unique(samples$group))</code></pre>
<pre class="r"><code>fh &lt;- round(6 + nrow(samples) / 15, 0)</code></pre>
<div id="alignment-statistics" class="section level1">
<h1>Alignment Statistics</h1>
<pre class="r"><code>alnFiles &lt;- here::here() %&gt;%
  list.files(recursive = TRUE, pattern = &quot;Log.final.out&quot;) 
alnStats &lt;- alnFiles %&gt;%
  lapply(function(x){
    importNgsLogs(x, type = &quot;star&quot;) %&gt;%
      mutate(Filename = x)
  }) %&gt;%
  bind_rows() %&gt;%
  mutate(Filename = basename(dirname(Filename))) %&gt;%
  left_join(samples, by = &quot;Filename&quot;) %&gt;%
  as.data.frame()</code></pre>
<ul>
<li>Across all files the total alignment rate ranged between <em>91.2%</em> and <em>96.0%</em></li>
<li>Uniquely aligned reads ranged between <em>79.5%</em> and <em>85.5%</em></li>
<li>The percentages of mapped reads which aligned to ‘too many’ locations and were discarded was between <em>0.22%</em> and <em>0.39%</em></li>
</ul>
<pre class="r"><code>ggplotly(
  alnStats %&gt;%
    dplyr::select(
      Filename, group, contains(&quot;Percent&quot;), -Total_Mapped_Percent
    ) %&gt;%
    mutate(
      Unmapped = Percent_Of_Reads_Unmapped_Too_Many_Mismatches +
        Percent_Of_Reads_Unmapped_Too_Short +
        Percent_Of_Reads_Unmapped_Other
    ) %&gt;%
    dplyr::select(
      Filename, group, contains(&quot;Mapped&quot;, ignore.case = FALSE), Unmapped
    ) %&gt;%
    pivot_longer(
      cols = contains(&quot;mapped&quot;),
      names_to = &quot;Category&quot;,
      values_to = &quot;Percent&quot;
    ) %&gt;%
    mutate(
      Category = str_remove_all(Category, &quot;(_Percent|Percent_Of_Reads_)&quot;),
      Category = str_replace_all(Category, &quot;_&quot;, &quot; &quot;),
      Category = as.factor(Category),
      Category = relevel(Category, ref = &quot;Uniquely Mapped Reads&quot;),
      Category = fct_rev(Category)
    ) %&gt;%
    ggplot(aes(Filename, Percent, fill = Category)) +
    geom_col(colour = &quot;black&quot;, size = 0.1) +
    facet_grid(~group, scales = &quot;free_x&quot;, space = &quot;free_x&quot;) +
    scale_y_continuous(expand = expansion(c(0, 0.05))) +
    scale_fill_viridis_d(option = &quot;E&quot;, direction = -1) +
    theme(
      axis.text.x = element_blank(),
      axis.ticks.x = element_blank()
    )
)</code></pre>
<div class="figure" style="text-align: center">
<div id="htmlwidget-895b7d97228efddd6387" style="width:768px;height:576px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-895b7d97228efddd6387">{"x":{"data":[{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[91.49,95.12,95.77,93.41],"x":[1,2,3,4],"y":[8.51000000000001,4.89,4.23,6.58999999999999],"text":["Filename: Gar15-13DHT1.R1.combined<br />Percent:  8.51<br />Category: Unmapped","Filename: Gar15-13DHT2.R1.combined<br />Percent:  4.89<br />Category: Unmapped","Filename: Gar15-13DHT3.R1.combined<br />Percent:  4.23<br />Category: Unmapped","Filename: Gar15-13DHT4.R1.combined<br />Percent:  6.59<br />Category: Unmapped"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(255,234,70,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Unmapped","legendgroup":"Unmapped","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[95.13,95.16,95.63,96.26,95.89],"x":[1,2,3,4,5],"y":[4.86,4.84999999999999,4.37,3.75,4.10000000000001],"text":["Filename: Gar15-13Sarm1.R1.combined<br />Percent:  4.86<br />Category: Unmapped","Filename: Gar15-13Sarm2.R1.combined<br />Percent:  4.85<br />Category: Unmapped","Filename: Gar15-13Sarm3.R1.combined<br />Percent:  4.37<br />Category: Unmapped","Filename: Gar15-13Sarm4.R1.combined<br />Percent:  3.75<br />Category: Unmapped","Filename: Gar15-13Sarm5.R1.combined<br />Percent:  4.10<br />Category: Unmapped"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(255,234,70,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Unmapped","legendgroup":"Unmapped","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[94.73,95.66,94.86,95.67,93.89],"x":[1,2,3,4,5],"y":[5.26000000000001,4.34,5.14,4.33,6.11],"text":["Filename: Gar15-13Veh1.R1.combined<br />Percent:  5.26<br />Category: Unmapped","Filename: Gar15-13Veh2.R1.combined<br />Percent:  4.34<br />Category: Unmapped","Filename: Gar15-13Veh3.R1.combined<br />Percent:  5.14<br />Category: Unmapped","Filename: Gar15-13Veh4.R1.combined<br />Percent:  4.33<br />Category: Unmapped","Filename: Gar15-13Veh5.R1.combined<br />Percent:  6.11<br />Category: Unmapped"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(255,234,70,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Unmapped","legendgroup":"Unmapped","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[91.16,94.86,95.45,93.12],"x":[1,2,3,4],"y":[0.329999999999998,0.259999999999991,0.319999999999993,0.290000000000006],"text":["Filename: Gar15-13DHT1.R1.combined<br />Percent:  0.33<br />Category: Mapped To Too Many Loci","Filename: Gar15-13DHT2.R1.combined<br />Percent:  0.26<br />Category: Mapped To Too Many Loci","Filename: Gar15-13DHT3.R1.combined<br />Percent:  0.32<br />Category: Mapped To Too Many Loci","Filename: Gar15-13DHT4.R1.combined<br />Percent:  0.29<br />Category: Mapped To Too Many Loci"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(166,157,117,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Mapped To Too Many Loci","legendgroup":"Mapped To Too Many Loci","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[94.87,94.83,95.41,95.96,95.61],"x":[1,2,3,4,5],"y":[0.260000000000005,0.329999999999998,0.219999999999999,0.299999999999997,0.280000000000001],"text":["Filename: Gar15-13Sarm1.R1.combined<br />Percent:  0.26<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Sarm2.R1.combined<br />Percent:  0.33<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Sarm3.R1.combined<br />Percent:  0.22<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Sarm4.R1.combined<br />Percent:  0.30<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Sarm5.R1.combined<br />Percent:  0.28<br />Category: Mapped To Too Many Loci"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(166,157,117,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Mapped To Too Many Loci","legendgroup":"Mapped To Too Many Loci","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[94.46,95.35,94.61,95.28,93.61],"x":[1,2,3,4,5],"y":[0.269999999999996,0.310000000000002,0.25,0.390000000000001,0.280000000000001],"text":["Filename: Gar15-13Veh1.R1.combined<br />Percent:  0.27<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Veh2.R1.combined<br />Percent:  0.31<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Veh3.R1.combined<br />Percent:  0.25<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Veh4.R1.combined<br />Percent:  0.39<br />Category: Mapped To Too Many Loci","Filename: Gar15-13Veh5.R1.combined<br />Percent:  0.28<br />Category: Mapped To Too Many Loci"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(166,157,117,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Mapped To Too Many Loci","legendgroup":"Mapped To Too Many Loci","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[80.74,84.52,83.47,79.54],"x":[1,2,3,4],"y":[10.42,10.34,11.98,13.58],"text":["Filename: Gar15-13DHT1.R1.combined<br />Percent: 10.42<br />Category: Mapped To Multiple Loci","Filename: Gar15-13DHT2.R1.combined<br />Percent: 10.34<br />Category: Mapped To Multiple Loci","Filename: Gar15-13DHT3.R1.combined<br />Percent: 11.98<br />Category: Mapped To Multiple Loci","Filename: Gar15-13DHT4.R1.combined<br />Percent: 13.58<br />Category: Mapped To Multiple Loci"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(87,92,109,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Mapped To Multiple Loci","legendgroup":"Mapped To Multiple Loci","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[83.27,85.15,81.44,85.46,81.2],"x":[1,2,3,4,5],"y":[11.6,9.68000000000001,13.97,10.5,14.41],"text":["Filename: Gar15-13Sarm1.R1.combined<br />Percent: 11.60<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Sarm2.R1.combined<br />Percent:  9.68<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Sarm3.R1.combined<br />Percent: 13.97<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Sarm4.R1.combined<br />Percent: 10.50<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Sarm5.R1.combined<br />Percent: 14.41<br />Category: Mapped To Multiple Loci"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(87,92,109,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Mapped To Multiple Loci","legendgroup":"Mapped To Multiple Loci","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[84.77,85.42,83.92,82.41,81.71],"x":[1,2,3,4,5],"y":[9.69,9.92999999999999,10.69,12.87,11.9],"text":["Filename: Gar15-13Veh1.R1.combined<br />Percent:  9.69<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Veh2.R1.combined<br />Percent:  9.93<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Veh3.R1.combined<br />Percent: 10.69<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Veh4.R1.combined<br />Percent: 12.87<br />Category: Mapped To Multiple Loci","Filename: Gar15-13Veh5.R1.combined<br />Percent: 11.90<br />Category: Mapped To Multiple Loci"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(87,92,109,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Mapped To Multiple Loci","legendgroup":"Mapped To Multiple Loci","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[0,0,0,0],"x":[1,2,3,4],"y":[80.74,84.52,83.47,79.54],"text":["Filename: Gar15-13DHT1.R1.combined<br />Percent: 80.74<br />Category: Uniquely Mapped Reads","Filename: Gar15-13DHT2.R1.combined<br />Percent: 84.52<br />Category: Uniquely Mapped Reads","Filename: Gar15-13DHT3.R1.combined<br />Percent: 83.47<br />Category: Uniquely Mapped Reads","Filename: Gar15-13DHT4.R1.combined<br />Percent: 79.54<br />Category: Uniquely Mapped Reads"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,32,77,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Uniquely Mapped Reads","legendgroup":"Uniquely Mapped Reads","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[0,0,0,0,0],"x":[1,2,3,4,5],"y":[83.27,85.15,81.44,85.46,81.2],"text":["Filename: Gar15-13Sarm1.R1.combined<br />Percent: 83.27<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Sarm2.R1.combined<br />Percent: 85.15<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Sarm3.R1.combined<br />Percent: 81.44<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Sarm4.R1.combined<br />Percent: 85.46<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Sarm5.R1.combined<br />Percent: 81.20<br />Category: Uniquely Mapped Reads"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,32,77,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Uniquely Mapped Reads","legendgroup":"Uniquely Mapped Reads","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[0,0,0,0,0],"x":[1,2,3,4,5],"y":[84.77,85.42,83.92,82.41,81.71],"text":["Filename: Gar15-13Veh1.R1.combined<br />Percent: 84.77<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Veh2.R1.combined<br />Percent: 85.42<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Veh3.R1.combined<br />Percent: 83.92<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Veh4.R1.combined<br />Percent: 82.41<br />Category: Uniquely Mapped Reads","Filename: Gar15-13Veh5.R1.combined<br />Percent: 81.71<br />Category: Uniquely Mapped Reads"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,32,77,1)","line":{"width":0.377952755905512,"color":"rgba(0,0,0,1)"}},"name":"Uniquely Mapped Reads","legendgroup":"Uniquely Mapped Reads","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":37.4307458143075,"r":18.9954337899543,"b":28.0060882800609,"l":43.1050228310502},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,0.32976598173516],"automargin":true,"type":"linear","autorange":false,"range":[0.4,4.6],"tickmode":"array","ticktext":["Gar15-13DHT1.R1.combined","Gar15-13DHT2.R1.combined","Gar15-13DHT3.R1.combined","Gar15-13DHT4.R1.combined"],"tickvals":[1,2,3,4],"categoryorder":"array","categoryarray":["Gar15-13DHT1.R1.combined","Gar15-13DHT2.R1.combined","Gar15-13DHT3.R1.combined","Gar15-13DHT4.R1.combined"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"annotations":[{"text":"Filename","x":0.5,"y":-0.00713470319634703,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"top","annotationType":"axis"},{"text":"Percent","x":-0.0139126712328767,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"DHT","x":0.16488299086758,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Sarm","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Veh","x":0.83511700913242,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Category","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,105.0105],"tickmode":"array","ticktext":["0","25","50","75","100"],"tickvals":[0,25,50,75,100],"categoryorder":"array","categoryarray":["0","25","50","75","100"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":0.32976598173516,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":0.32976598173516,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.336900684931507,"x1":0.663099315068493,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.336900684931507,"x1":0.663099315068493,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.67023401826484,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.67023401826484,"x1":1,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0.4,5.6],"tickmode":"array","ticktext":["Gar15-13Sarm1.R1.combined","Gar15-13Sarm2.R1.combined","Gar15-13Sarm3.R1.combined","Gar15-13Sarm4.R1.combined","Gar15-13Sarm5.R1.combined"],"tickvals":[1,2,3,4,5],"categoryorder":"array","categoryarray":["Gar15-13Sarm1.R1.combined","Gar15-13Sarm2.R1.combined","Gar15-13Sarm3.R1.combined","Gar15-13Sarm4.R1.combined","Gar15-13Sarm5.R1.combined"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"domain":[0.336900684931507,0.663099315068493],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"xaxis3":{"type":"linear","autorange":false,"range":[0.4,5.6],"tickmode":"array","ticktext":["Gar15-13Veh1.R1.combined","Gar15-13Veh2.R1.combined","Gar15-13Veh3.R1.combined","Gar15-13Veh4.R1.combined","Gar15-13Veh5.R1.combined"],"tickvals":[1,2,3,4,5],"categoryorder":"array","categoryarray":["Gar15-13Veh1.R1.combined","Gar15-13Veh2.R1.combined","Gar15-13Veh3.R1.combined","Gar15-13Veh4.R1.combined","Gar15-13Veh5.R1.combined"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"domain":[0.67023401826484,1],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.972933070866142},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"7da91948ca21":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"7da91948ca21","visdat":{"7da91948ca21":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
<em>Alignment rates across all libraries</em>
</p>
</div>
</div>
<div id="read-assignment-to-genes" class="section level1">
<h1>Read Assignment To Genes</h1>
<div id="annotation-setup" class="section level3">
<h3>Annotation Setup</h3>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, as.character(config$ref$release))) %&gt;%
  subset(genome == config$ref$build)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- genes(ensDb)
transGR &lt;- transcripts(ensDb)</code></pre>
<pre class="r"><code>mcols(transGR) &lt;- mcols(transGR) %&gt;%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c(&quot;nexon&quot;, &quot;tx_len&quot;)]
  )</code></pre>
<pre class="r"><code>mcols(genesGR) &lt;- mcols(genesGR) %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %&gt;%
  left_join(
    mcols(transGR) %&gt;%
      as.data.frame() %&gt;%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %&gt;%
      group_by(gene_id) %&gt;%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %&gt;%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = &quot;_&quot;)
      ),
    by = &quot;gene_id&quot;
  ) %&gt;%
  set_rownames(.$gene_id) %&gt;%
  as(&quot;DataFrame&quot;)</code></pre>
<p>Annotation data was loaded as an <code>EnsDb</code> object, using Ensembl release 101. Transcript level gene lengths and GC content was converted to gene level values using:</p>
<ul>
<li><strong>GC Content</strong>: The total GC content divided by the total length of transcripts</li>
<li><strong>Gene Length</strong>: The mean transcript length</li>
</ul>
<pre class="r"><code>write_rds(genesGR, here::here(&quot;output/genesGR.rds&quot;), compress = &quot;gz&quot;)</code></pre>
</div>
<div id="counts" class="section level2">
<h2>Counts</h2>
<pre class="r"><code>countSummary &lt;- here::here(&quot;data/aligned/counts/counts.out.summary&quot;) %&gt;%
  importNgsLogs(type = &quot;featureCounts&quot;) %&gt;%
  mutate(
    Total = rowSums(
      dplyr::select_if(., is.numeric)
    ),
    Filename = basename(dirname(Sample))
  ) %&gt;%
  dplyr::select(-Sample) %&gt;%
  left_join(samples)</code></pre>
<p>Read assignment to genes was performed using Ensembl release 101 which used the genome build GRCh38 for generation of gene models. When assigning reads to genes, <code>featureCounts</code> was run setting the following criteria:</p>
<ul>
<li>Libraries were assumed to be <em>unstranded</em></li>
<li>The <em>minimum percentage</em> of a read which needed to overlap an exon before being counted was 90%</li>
<li>In addition to the minimum percentage, <em>a minimum of 35 bases</em> must also overlap an exon before a read is counted</li>
<li>The <em>minimum alignment quality score</em> for a read to be counted was 1</li>
<li>Counting of multi-mapped reads was permitted using fractional counts</li>
</ul>
<p>Using these settings:</p>
<ul>
<li>The percentages of reads assigned to genes ranged between <em>38%</em> and <em>52%</em></li>
<li>Of the total reads:
<ul>
<li>Between <em>0%</em> and <em>0%</em> were unassigned due to multi-mapping</li>
<li>Between <em>11%</em> and <em>26%</em> were aligned but didn’t overlap any known genes</li>
<li>Between <em>1.22%</em> and <em>1.98%</em> were unassigned as they failed the proportion overlapping criteria</li>
<li>Between <em>2.05%</em> and <em>2.80%</em> were unassigned as they were considered ambiguous</li>
</ul></li>
</ul>
<pre class="r"><code>ggplotly(
  countSummary %&gt;%
    pivot_longer(
      cols = contains(&quot;assigned&quot;),
      names_to = &quot;Status&quot;,
      values_to = &quot;Reads&quot;
    ) %&gt;%
    dplyr::filter(Reads &gt; 0) %&gt;%
    mutate(
      Percent = round(100 * Reads / Total, 2)
    ) %&gt;%
    arrange(Percent) %&gt;%
    mutate(
      Status = str_replace_all(Status, &quot;Unassigned_&quot;, &quot;Unassigned: &quot;),
      Status = str_replace_all(Status, &quot;_&quot;, &quot; &quot;),
      Status = fct_inorder(Status)) %&gt;%
    ggplot(aes(sample, Percent, fill = Status)) +
    geom_col() +
    facet_grid(~group, scales = &quot;free_x&quot;, space = &quot;free_x&quot;) +
    scale_fill_viridis_d(option = &quot;E&quot;, direction = -1) +
    scale_y_continuous(
      labels = ngsReports:::.addPercent,
      expand = expansion(c(0, 0.05))
    ) + 
    theme(
      axis.text.x = element_blank(),
      axis.ticks.x = element_blank()
    )
)</code></pre>
<div class="figure" style="text-align: center">
<div id="htmlwidget-f9a20bb858604fdeefb0" style="width:768px;height:576px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-f9a20bb858604fdeefb0">{"x":{"data":[{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[98.4,98.4,98.29,98.02],"x":[3,4,2,1],"y":[1.60000000000001,1.61,1.70999999999999,1.98],"text":["sample: Gar15-13DHT3<br />Percent:  1.60<br />Status: Unassigned: Overlapping Length","sample: Gar15-13DHT4<br />Percent:  1.61<br />Status: Unassigned: Overlapping Length","sample: Gar15-13DHT2<br />Percent:  1.71<br />Status: Unassigned: Overlapping Length","sample: Gar15-13DHT1<br />Percent:  1.98<br />Status: Unassigned: Overlapping Length"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(255,234,70,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: Overlapping Length","legendgroup":"Unassigned: Overlapping Length","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[98.79,98.75,98.44,98.43,98.24],"x":[3,5,4,1,2],"y":[1.22,1.25,1.55,1.57000000000001,1.76000000000001],"text":["sample: Gar15-13Sarm3<br />Percent:  1.22<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Sarm5<br />Percent:  1.25<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Sarm4<br />Percent:  1.55<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Sarm1<br />Percent:  1.57<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Sarm2<br />Percent:  1.76<br />Status: Unassigned: Overlapping Length"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(255,234,70,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: Overlapping Length","legendgroup":"Unassigned: Overlapping Length","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[98.55,98.43,98.38,98.28,98.26],"x":[4,3,5,1,2],"y":[1.44,1.56999999999999,1.61,1.73,1.75],"text":["sample: Gar15-13Veh4<br />Percent:  1.44<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Veh3<br />Percent:  1.57<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Veh5<br />Percent:  1.61<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Veh1<br />Percent:  1.73<br />Status: Unassigned: Overlapping Length","sample: Gar15-13Veh2<br />Percent:  1.75<br />Status: Unassigned: Overlapping Length"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(255,234,70,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: Overlapping Length","legendgroup":"Unassigned: Overlapping Length","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[96.27,96.16,95.76,95.79],"x":[4,3,1,2],"y":[2.13,2.23999999999999,2.25999999999999,2.5],"text":["sample: Gar15-13DHT4<br />Percent:  2.13<br />Status: Unassigned: Ambiguity","sample: Gar15-13DHT3<br />Percent:  2.24<br />Status: Unassigned: Ambiguity","sample: Gar15-13DHT1<br />Percent:  2.26<br />Status: Unassigned: Ambiguity","sample: Gar15-13DHT2<br />Percent:  2.50<br />Status: Unassigned: Ambiguity"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(188,175,111,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: Ambiguity","legendgroup":"Unassigned: Ambiguity","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[96.7,95.93,96.03,96.27,95.69],"x":[5,2,1,3,4],"y":[2.05,2.30999999999999,2.39999999999999,2.52,2.75],"text":["sample: Gar15-13Sarm5<br />Percent:  2.05<br />Status: Unassigned: Ambiguity","sample: Gar15-13Sarm2<br />Percent:  2.31<br />Status: Unassigned: Ambiguity","sample: Gar15-13Sarm1<br />Percent:  2.40<br />Status: Unassigned: Ambiguity","sample: Gar15-13Sarm3<br />Percent:  2.52<br />Status: Unassigned: Ambiguity","sample: Gar15-13Sarm4<br />Percent:  2.75<br />Status: Unassigned: Ambiguity"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(188,175,111,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: Ambiguity","legendgroup":"Unassigned: Ambiguity","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[96.27,96.04,95.88,95.57,95.48],"x":[4,5,3,2,1],"y":[2.28,2.34,2.55,2.69,2.8],"text":["sample: Gar15-13Veh4<br />Percent:  2.28<br />Status: Unassigned: Ambiguity","sample: Gar15-13Veh5<br />Percent:  2.34<br />Status: Unassigned: Ambiguity","sample: Gar15-13Veh3<br />Percent:  2.55<br />Status: Unassigned: Ambiguity","sample: Gar15-13Veh2<br />Percent:  2.69<br />Status: Unassigned: Ambiguity","sample: Gar15-13Veh1<br />Percent:  2.80<br />Status: Unassigned: Ambiguity"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(188,175,111,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: Ambiguity","legendgroup":"Unassigned: Ambiguity","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[81.65,78.74,76.49,72.8],"x":[4,3,2,1],"y":[14.62,17.42,19.3,22.96],"text":["sample: Gar15-13DHT4<br />Percent: 14.62<br />Status: Unassigned: NoFeatures","sample: Gar15-13DHT3<br />Percent: 17.42<br />Status: Unassigned: NoFeatures","sample: Gar15-13DHT2<br />Percent: 19.30<br />Status: Unassigned: NoFeatures","sample: Gar15-13DHT1<br />Percent: 22.96<br />Status: Unassigned: NoFeatures"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(124,123,120,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: NoFeatures","legendgroup":"Unassigned: NoFeatures","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[84.88,80.86,79.77,76.73,69.76],"x":[3,5,4,1,2],"y":[11.39,15.84,15.92,19.3,26.17],"text":["sample: Gar15-13Sarm3<br />Percent: 11.39<br />Status: Unassigned: NoFeatures","sample: Gar15-13Sarm5<br />Percent: 15.84<br />Status: Unassigned: NoFeatures","sample: Gar15-13Sarm4<br />Percent: 15.92<br />Status: Unassigned: NoFeatures","sample: Gar15-13Sarm1<br />Percent: 19.30<br />Status: Unassigned: NoFeatures","sample: Gar15-13Sarm2<br />Percent: 26.17<br />Status: Unassigned: NoFeatures"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(124,123,120,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: NoFeatures","legendgroup":"Unassigned: NoFeatures","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[79.51,77.62,76.69,74.57,74.18],"x":[4,3,5,1,2],"y":[16.76,18.26,19.35,20.91,21.39],"text":["sample: Gar15-13Veh4<br />Percent: 16.76<br />Status: Unassigned: NoFeatures","sample: Gar15-13Veh3<br />Percent: 18.26<br />Status: Unassigned: NoFeatures","sample: Gar15-13Veh5<br />Percent: 19.35<br />Status: Unassigned: NoFeatures","sample: Gar15-13Veh1<br />Percent: 20.91<br />Status: Unassigned: NoFeatures","sample: Gar15-13Veh2<br />Percent: 21.39<br />Status: Unassigned: NoFeatures"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(124,123,120,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: NoFeatures","legendgroup":"Unassigned: NoFeatures","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[50.45,45.31,46.12,43.14],"x":[2,1,3,4],"y":[26.04,27.49,32.62,38.51],"text":["sample: Gar15-13DHT2<br />Percent: 26.04<br />Status: Unassigned: MappingQuality","sample: Gar15-13DHT1<br />Percent: 27.49<br />Status: Unassigned: MappingQuality","sample: Gar15-13DHT3<br />Percent: 32.62<br />Status: Unassigned: MappingQuality","sample: Gar15-13DHT4<br />Percent: 38.51<br />Status: Unassigned: MappingQuality"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(65,77,107,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: MappingQuality","legendgroup":"Unassigned: MappingQuality","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[45.76,51.98,45.33,43.75,38.29],"x":[2,4,1,3,5],"y":[24,27.79,31.4,41.13,42.57],"text":["sample: Gar15-13Sarm2<br />Percent: 24.00<br />Status: Unassigned: MappingQuality","sample: Gar15-13Sarm4<br />Percent: 27.79<br />Status: Unassigned: MappingQuality","sample: Gar15-13Sarm1<br />Percent: 31.40<br />Status: Unassigned: MappingQuality","sample: Gar15-13Sarm3<br />Percent: 41.13<br />Status: Unassigned: MappingQuality","sample: Gar15-13Sarm5<br />Percent: 42.57<br />Status: Unassigned: MappingQuality"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(65,77,107,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: MappingQuality","legendgroup":"Unassigned: MappingQuality","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[51.38,50.27,49.09,44.65,43.19],"x":[1,2,3,5,4],"y":[23.19,23.91,28.53,32.04,36.32],"text":["sample: Gar15-13Veh1<br />Percent: 23.19<br />Status: Unassigned: MappingQuality","sample: Gar15-13Veh2<br />Percent: 23.91<br />Status: Unassigned: MappingQuality","sample: Gar15-13Veh3<br />Percent: 28.53<br />Status: Unassigned: MappingQuality","sample: Gar15-13Veh5<br />Percent: 32.04<br />Status: Unassigned: MappingQuality","sample: Gar15-13Veh4<br />Percent: 36.32<br />Status: Unassigned: MappingQuality"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(65,77,107,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Unassigned: MappingQuality","legendgroup":"Unassigned: MappingQuality","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9],"base":[0,0,0,0],"x":[4,1,3,2],"y":[43.14,45.31,46.12,50.45],"text":["sample: Gar15-13DHT4<br />Percent: 43.14<br />Status: Assigned","sample: Gar15-13DHT1<br />Percent: 45.31<br />Status: Assigned","sample: Gar15-13DHT3<br />Percent: 46.12<br />Status: Assigned","sample: Gar15-13DHT2<br />Percent: 50.45<br />Status: Assigned"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,32,77,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Assigned","legendgroup":"Assigned","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[0,0,0,0,0],"x":[5,3,1,2,4],"y":[38.29,43.75,45.33,45.76,51.98],"text":["sample: Gar15-13Sarm5<br />Percent: 38.29<br />Status: Assigned","sample: Gar15-13Sarm3<br />Percent: 43.75<br />Status: Assigned","sample: Gar15-13Sarm1<br />Percent: 45.33<br />Status: Assigned","sample: Gar15-13Sarm2<br />Percent: 45.76<br />Status: Assigned","sample: Gar15-13Sarm4<br />Percent: 51.98<br />Status: Assigned"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,32,77,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Assigned","legendgroup":"Assigned","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"v","width":[0.9,0.9,0.9,0.9,0.9],"base":[0,0,0,0,0],"x":[4,5,3,2,1],"y":[43.19,44.65,49.09,50.27,51.38],"text":["sample: Gar15-13Veh4<br />Percent: 43.19<br />Status: Assigned","sample: Gar15-13Veh5<br />Percent: 44.65<br />Status: Assigned","sample: Gar15-13Veh3<br />Percent: 49.09<br />Status: Assigned","sample: Gar15-13Veh2<br />Percent: 50.27<br />Status: Assigned","sample: Gar15-13Veh1<br />Percent: 51.38<br />Status: Assigned"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(0,32,77,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"Assigned","legendgroup":"Assigned","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":37.4307458143075,"r":18.9954337899543,"b":28.0060882800609,"l":48.9497716894977},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,0.32976598173516],"automargin":true,"type":"linear","autorange":false,"range":[0.4,4.6],"tickmode":"array","ticktext":["Gar15-13DHT1","Gar15-13DHT2","Gar15-13DHT3","Gar15-13DHT4"],"tickvals":[1,2,3,4],"categoryorder":"array","categoryarray":["Gar15-13DHT1","Gar15-13DHT2","Gar15-13DHT3","Gar15-13DHT4"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"annotations":[{"text":"sample","x":0.5,"y":-0.00713470319634703,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"top","annotationType":"axis"},{"text":"Percent","x":-0.0167665525114155,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"DHT","x":0.16488299086758,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Sarm","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Veh","x":0.83511700913242,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Status","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,105.0105],"tickmode":"array","ticktext":["0%","25%","50%","75%","100%"],"tickvals":[0,25,50,75,100],"categoryorder":"array","categoryarray":["0%","25%","50%","75%","100%"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":0.32976598173516,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":0.32976598173516,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.336900684931507,"x1":0.663099315068493,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.336900684931507,"x1":0.663099315068493,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.67023401826484,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.67023401826484,"x1":1,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0.4,5.6],"tickmode":"array","ticktext":["Gar15-13Sarm1","Gar15-13Sarm2","Gar15-13Sarm3","Gar15-13Sarm4","Gar15-13Sarm5"],"tickvals":[1,2,3,4,5],"categoryorder":"array","categoryarray":["Gar15-13Sarm1","Gar15-13Sarm2","Gar15-13Sarm3","Gar15-13Sarm4","Gar15-13Sarm5"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"domain":[0.336900684931507,0.663099315068493],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"xaxis3":{"type":"linear","autorange":false,"range":[0.4,5.6],"tickmode":"array","ticktext":["Gar15-13Veh1","Gar15-13Veh2","Gar15-13Veh3","Gar15-13Veh4","Gar15-13Veh5"],"tickvals":[1,2,3,4,5],"categoryorder":"array","categoryarray":["Gar15-13Veh1","Gar15-13Veh2","Gar15-13Veh3","Gar15-13Veh4","Gar15-13Veh5"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"domain":[0.67023401826484,1],"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.972933070866142},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"7da958ef8aaa":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"7da958ef8aaa","visdat":{"7da958ef8aaa":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
<em>Rate of mapped reads being assigned to genes</em>
</p>
</div>
</div>
<div id="total-detected-genes" class="section level2">
<h2>Total Detected Genes</h2>
<pre class="r"><code>counts &lt;- here::here(&quot;data/aligned/counts/counts.out&quot;) %&gt;%
  read_tsv(comment = &quot;#&quot;) %&gt;%
  rename_all(str_remove_all, pattern = &quot;/Aligned.+&quot;) %&gt;%
  rename_all(basename) %&gt;%
  dplyr::select(-Chr, -Start, -End, -Strand, -Length) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  as.matrix()</code></pre>
<ul>
<li>Of the 60,671 genes defined in this annotation build, 19,572 genes had no reads assigned in any samples.</li>
<li>The numbers of genes with at least one read assigned ranged between <em>28,556</em> and <em>31,871</em> across all samples.</li>
</ul>
<pre class="r"><code>counts %&gt;%
  as_tibble() %&gt;%
  mutate(
    across(everything(), as.logical)
  ) %&gt;%
  summarise(
    across(everything(), sum)
  ) %&gt;%
  pivot_longer(
    everything(), names_to = &quot;Filename&quot;, values_to = &quot;Detected&quot;
  ) %&gt;%
  left_join(samples) %&gt;%
  ggplot(aes(group, Detected, colour = group)) +
  geom_point() +
  geom_segment(
    aes(xend = group, y = 0, yend = Detected),
    data = . %&gt;% 
      group_by(group) %&gt;%
      summarise(Detected = min(Detected)),
    colour = &quot;black&quot;, size = 1/4) +
  scale_y_continuous(labels = comma, expand = expansion(c(0, 0.05))) +
  scale_colour_manual(values = group_cols) +
  labs(
    x = &quot;Group&quot;, 
    y = &quot;Genes Detected&quot;,
    colour = &quot;Group&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/qc_aligned.Rmd/plotDetectedGenes-1.png" alt="*Total numbers of genes detected across all samples and groups.*" width="768" />
<p class="caption">
<em>Total numbers of genes detected across all samples and groups.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotDetectedGenes-1">
Past versions of plotDetectedGenes-1.png
</button>
</p>
<div id="fig-plotDetectedGenes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/blob/4ab5279ba0abcba184ad82430844d437f24445ed/docs/figure/qc_aligned.Rmd/plotDetectedGenes-1.png" target="_blank">4ab5279</a>
</td>
<td>
Steve Pederson
</td>
<td>
2021-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotly::ggplotly(
  counts %&gt;%
    is_greater_than(0) %&gt;%
    rowSums() %&gt;%
    table() %&gt;%
    enframe(name = &quot;n_samples&quot;, value = &quot;n_genes&quot;) %&gt;%
    mutate(
      n_samples = as.integer(n_samples),
      n_genes = as.integer(n_genes),
    ) %&gt;%
    arrange(desc(n_samples)) %&gt;%
    mutate(
      Detectable = cumsum(n_genes),
      Undetectable = sum(n_genes) - Detectable
    ) %&gt;%
    pivot_longer(
      cols = ends_with(&quot;table&quot;),
      names_to = &quot;Status&quot;,
      values_to = &quot;Number of Genes&quot;
    ) %&gt;%
    dplyr::rename(
      `Number of Samples` = n_samples,
    ) %&gt;%
    ggplot(aes(`Number of Samples`, `Number of Genes`, colour = Status)) +
    geom_line() +
    geom_vline(
      aes(xintercept = `Mean Sample Number`),
      data = . %&gt;%
        summarise(`Mean Sample Number` = mean(`Number of Samples`)),
      linetype = 2,
      colour = &quot;grey50&quot;
    ) +
    scale_x_continuous(expand = expansion(c(0.01, 0.01))) +
    scale_y_continuous(labels = comma) +
    scale_colour_manual(values = c(rgb(0.1, 0.7, 0.2), rgb(0.7, 0.1, 0.1))) +
    labs(
      x = &quot;Samples &gt; 0&quot;
    )
)</code></pre>
<div class="figure" style="text-align: center">
<div id="htmlwidget-f6d715ce67a4a530071a" style="width:768px;height:576px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-f6d715ce67a4a530071a">{"x":{"data":[{"x":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14],"y":[60671,41099,37442,35147,33522,32228,31034,29922,28926,27917,26906,25868,24763,23467,21580],"text":["Number of Samples:  0<br />Number of Genes: 60671<br />Status: Detectable","Number of Samples:  1<br />Number of Genes: 41099<br />Status: Detectable","Number of Samples:  2<br />Number of Genes: 37442<br />Status: Detectable","Number of Samples:  3<br />Number of Genes: 35147<br />Status: Detectable","Number of Samples:  4<br />Number of Genes: 33522<br />Status: Detectable","Number of Samples:  5<br />Number of Genes: 32228<br />Status: Detectable","Number of Samples:  6<br />Number of Genes: 31034<br />Status: Detectable","Number of Samples:  7<br />Number of Genes: 29922<br />Status: Detectable","Number of Samples:  8<br />Number of Genes: 28926<br />Status: Detectable","Number of Samples:  9<br />Number of Genes: 27917<br />Status: Detectable","Number of Samples: 10<br />Number of Genes: 26906<br />Status: Detectable","Number of Samples: 11<br />Number of Genes: 25868<br />Status: Detectable","Number of Samples: 12<br />Number of Genes: 24763<br />Status: Detectable","Number of Samples: 13<br />Number of Genes: 23467<br />Status: Detectable","Number of Samples: 14<br />Number of Genes: 21580<br />Status: Detectable"],"type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(26,179,51,1)","dash":"solid"},"hoveron":"points","name":"Detectable","legendgroup":"Detectable","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14],"y":[0,19572,23229,25524,27149,28443,29637,30749,31745,32754,33765,34803,35908,37204,39091],"text":["Number of Samples:  0<br />Number of Genes:     0<br />Status: Undetectable","Number of Samples:  1<br />Number of Genes: 19572<br />Status: Undetectable","Number of Samples:  2<br />Number of Genes: 23229<br />Status: Undetectable","Number of Samples:  3<br />Number of Genes: 25524<br />Status: Undetectable","Number of Samples:  4<br />Number of Genes: 27149<br />Status: Undetectable","Number of Samples:  5<br />Number of Genes: 28443<br />Status: Undetectable","Number of Samples:  6<br />Number of Genes: 29637<br />Status: Undetectable","Number of Samples:  7<br />Number of Genes: 30749<br />Status: Undetectable","Number of Samples:  8<br />Number of Genes: 31745<br />Status: Undetectable","Number of Samples:  9<br />Number of Genes: 32754<br />Status: Undetectable","Number of Samples: 10<br />Number of Genes: 33765<br />Status: Undetectable","Number of Samples: 11<br />Number of Genes: 34803<br />Status: Undetectable","Number of Samples: 12<br />Number of Genes: 35908<br />Status: Undetectable","Number of Samples: 13<br />Number of Genes: 37204<br />Status: Undetectable","Number of Samples: 14<br />Number of Genes: 39091<br />Status: Undetectable"],"type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(179,26,26,1)","dash":"solid"},"hoveron":"points","name":"Undetectable","legendgroup":"Undetectable","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[7,7],"y":[-3033.55,63704.55],"text":"Mean Sample Number: 7","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(127,127,127,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":25.7412480974125,"r":7.30593607305936,"b":39.6955859969559,"l":60.6392694063927},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.14,14.14],"tickmode":"array","ticktext":["0","5","10"],"tickvals":[0,5,10],"categoryorder":"array","categoryarray":["0","5","10"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"Samples > 0","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-3033.55,63704.55],"tickmode":"array","ticktext":["0","20,000","40,000","60,000"],"tickvals":[0,20000,40000,60000],"categoryorder":"array","categoryarray":["0","20,000","40,000","60,000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":{"text":"Number of Genes","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.972933070866142},"annotations":[{"text":"Status","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"7da932a3c2ba":{"x":{},"y":{},"colour":{},"type":"scatter"},"7da91bc74d6a":{"xintercept":{}}},"cur_data":"7da932a3c2ba","visdat":{"7da932a3c2ba":["function (y) ","x"],"7da91bc74d6a":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
<em>Total numbers of genes detected shown against the number of samples with at least one read assigned to each gene.</em>
</p>
</div>
</div>
<div id="library-sizes" class="section level2">
<h2>Library Sizes</h2>
<p>After assignment to genes, library sizes ranged between <em>10,449,789</em> and <em>17,211,954</em> reads, with a median library size of 12,678,980 reads.</p>
<pre class="r"><code>plotly::ggplotly(
  counts %&gt;%
    colSums() %&gt;%
    enframe(
      name = &quot;Filename&quot;, value = &quot;Library Size&quot;
    ) %&gt;%
    left_join(samples) %&gt;%
    ggplot(aes(group, `Library Size`, colour = group, label = Filename)) +
    geom_point() +
    geom_segment(
      aes(x = group, xend = group, y = 0, yend = `Library Size`),
      data = . %&gt;% 
        group_by(group) %&gt;%
        summarise(`Library Size` = min(`Library Size`)),
      colour = &quot;black&quot;, size = 1/4,
      inherit.aes = FALSE
    ) +
    scale_y_continuous(labels = comma, expand = expansion(c(0, 0.05))) +
    scale_colour_manual(values = group_cols) +
    labs(
      x = &quot;Group&quot;, 
      colour = &quot;Group&quot;
    )
)</code></pre>
<div class="figure" style="text-align: center">
<div id="htmlwidget-4186b9ff98d638e5b2e8" style="width:768px;height:576px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-4186b9ff98d638e5b2e8">{"x":{"data":[{"x":[1,1,1,1],"y":[10656777.92,12088864.05,13906470.32,12469905.67],"text":["Library Size: 10656778<br />Filename: Gar15-13DHT1.R1.combined","Library Size: 12088864<br />Filename: Gar15-13DHT2.R1.combined","Library Size: 13906470<br />Filename: Gar15-13DHT3.R1.combined","Library Size: 12469906<br />Filename: Gar15-13DHT4.R1.combined"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(59,153,177,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(59,153,177,1)"}},"hoveron":"points","name":"DHT","legendgroup":"DHT","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2,2,2,2,2],"y":[16708624.97,12086298.15,13603686.31,11950883.24,11953369.69],"text":["Library Size: 16708625<br />Filename: Gar15-13Sarm1.R1.combined","Library Size: 12086298<br />Filename: Gar15-13Sarm2.R1.combined","Library Size: 13603686<br />Filename: Gar15-13Sarm3.R1.combined","Library Size: 11950883<br />Filename: Gar15-13Sarm4.R1.combined","Library Size: 11953370<br />Filename: Gar15-13Sarm5.R1.combined"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(234,203,43,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(234,203,43,1)"}},"hoveron":"points","name":"Sarm","legendgroup":"Sarm","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3,3,3,3,3],"y":[17211954.03,12888054.82,13127235.8,10449789.28,15193927.82],"text":["Library Size: 17211954<br />Filename: Gar15-13Veh1.R1.combined","Library Size: 12888055<br />Filename: Gar15-13Veh2.R1.combined","Library Size: 13127236<br />Filename: Gar15-13Veh3.R1.combined","Library Size: 10449789<br />Filename: Gar15-13Veh4.R1.combined","Library Size: 15193928<br />Filename: Gar15-13Veh5.R1.combined"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(245,25,28,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(245,25,28,1)"}},"hoveron":"points","name":"Veh","legendgroup":"Veh","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1,1,null,2,2,null,3,3],"y":[0,10656777.92,null,0,11950883.24,null,0,10449789.28],"text":["y: 0<br />Library Size: 10656778","y: 0<br />Library Size: 10656778",null,"y: 0<br />Library Size: 11950883","y: 0<br />Library Size: 11950883",null,"y: 0<br />Library Size: 10449789","y: 0<br />Library Size: 10449789"],"type":"scatter","mode":"lines","line":{"width":0.94488188976378,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":25.7412480974125,"r":7.30593607305936,"b":39.6955859969559,"l":84.0182648401827},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,3.6],"tickmode":"array","ticktext":["DHT","Sarm","Veh"],"tickvals":[1,2,3],"categoryorder":"array","categoryarray":["DHT","Sarm","Veh"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"Group","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,18072551.7315],"tickmode":"array","ticktext":["0","5,000,000","10,000,000","15,000,000"],"tickvals":[0,5000000,10000000,15000000],"categoryorder":"array","categoryarray":["0","5,000,000","10,000,000","15,000,000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":{"text":"Library Size","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.972933070866142},"annotations":[{"text":"Group","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"7da94ecf57f8":{"x":{},"y":{},"colour":{},"label":{},"type":"scatter"},"7da959940187":{"x":{},"y":{},"xend":{},"yend":{}}},"cur_data":"7da94ecf57f8","visdat":{"7da94ecf57f8":["function (y) ","x"],"7da959940187":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
<em>Library sizes across all samples and groups</em>
</p>
</div>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<div id="sample-similarity" class="section level3">
<h3>Sample Similarity</h3>
<pre class="r"><code>pca &lt;- counts %&gt;%
  .[rowSums(. == 0) &lt; ncol(.)/2,] %&gt;%
  cpm(log = TRUE) %&gt;%
  t() %&gt;%
  prcomp()</code></pre>
<p>A PCA was performed using logCPM values from the subset of 28,926 genes with at least one read in more than half of the samples.</p>
<pre class="r"><code>showLabel &lt;- nrow(samples) &lt;= 20
pca %&gt;%
  autoplot(
    data = samples, 
    colour = &quot;group&quot;, 
    label = showLabel, 
    label.repel = showLabel, 
    label.show.legend = FALSE
  ) +
  labs(colour = &quot;Group&quot;) +
  scale_colour_manual(values = group_cols)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/qc_aligned.Rmd/plotPCA-1.png" alt="*PCA plot of all samples.*" width="768" />
<p class="caption">
<em>PCA plot of all samples.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotPCA-1">
Past versions of plotPCA-1.png
</button>
</p>
<div id="fig-plotPCA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/blob/4ab5279ba0abcba184ad82430844d437f24445ed/docs/figure/qc_aligned.Rmd/plotPCA-1.png" target="_blank">4ab5279</a>
</td>
<td>
Steve Pederson
</td>
<td>
2021-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc-and-length-biases" class="section level3">
<h3>GC and Length Biases</h3>
<pre class="r"><code>mcols(genesGR) %&gt;%
  as.data.frame() %&gt;%
  dplyr::filter(gene_id %in% rownames(pca$rotation)) %&gt;%
  as_tibble() %&gt;%
  mutate(
    bin_length = cut(
      x = ave_tx_len,
      labels = seq_len(10),
      breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin_gc = cut(
      x = gc_content,
      labels = seq_len(10),
      breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
      include.lowest = TRUE
    ),
    bin = paste(bin_gc, bin_length, sep = &quot;_&quot;)
  ) %&gt;%
  dplyr::select(gene_id, contains(&quot;bin&quot;)) %&gt;%
  mutate(
    PC1 = pca$rotation[gene_id, &quot;PC1&quot;],
    PC2 = pca$rotation[gene_id, &quot;PC2&quot;]
  ) %&gt;%
  pivot_longer(
    cols = c(&quot;PC1&quot;, &quot;PC2&quot;),
    names_to = &quot;PC&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  group_by(PC, bin_gc, bin_length, bin) %&gt;%
  summarise(
    Size = n(),
    mean = mean(value),
    sd = sd(value),
    t = t.test(value)$statistic,
    p = t.test(value)$p.value,
    adjP = p.adjust(p, method = &quot;bonf&quot;)
  ) %&gt;%
  ggplot(
    aes(bin_length, bin_gc, colour = t, alpha = -log10(adjP), size = Size)
  ) +
  geom_point() +
  facet_wrap(~PC) +
  scale_colour_gradient2() +
  scale_size_continuous(range = c(1, 10)) +
  labs(
    x = &quot;Average Transcript Length&quot;,
    y = &quot;GC Content&quot;,
    alpha = expression(paste(-log[10], p[adj]))) +
  theme(
    panel.grid = element_blank(),
    legend.position = &quot;bottom&quot;
    ) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/qc_aligned.Rmd/plotPCA_GC_Length-1.png" alt="*Contribution of each GC/Length Bin to PC1 and PC2. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values.*" width="768" />
<p class="caption">
<em>Contribution of each GC/Length Bin to PC1 and PC2. Fill colours indicate the t-statistic, with tranparency denoting significance as -log10(p), using Bonferroni-adjusted p-values.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotPCA_GC_Length-1">
Past versions of plotPCA_GC_Length-1.png
</button>
</p>
<div id="fig-plotPCA_GC_Length-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/PDX_Gar15-13_Veh_DHT_Sarm_RNASeq/blob/4ab5279ba0abcba184ad82430844d437f24445ed/docs/figure/qc_aligned.Rmd/plotPCA_GC_Length-1.png" target="_blank">4ab5279</a>
</td>
<td>
Steve Pederson
</td>
<td>
2021-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] magrittr_2.0.1          ensembldb_2.14.0        AnnotationFilter_1.14.0
 [4] GenomicFeatures_1.42.1  AnnotationDbi_1.52.0    Biobase_2.50.0         
 [7] GenomicRanges_1.42.0    GenomeInfoDb_1.26.2     IRanges_2.24.1         
[10] S4Vectors_0.28.1        AnnotationHub_2.22.0    BiocFileCache_1.14.0   
[13] dbplyr_2.0.0            ggfortify_0.4.11        edgeR_3.32.0           
[16] limma_3.46.0            plotly_4.9.3            glue_1.4.2             
[19] pander_0.6.3            scales_1.1.1            yaml_2.2.1             
[22] forcats_0.5.0           stringr_1.4.0           dplyr_1.0.2            
[25] purrr_0.3.4             readr_1.4.0             tidyr_1.1.2            
[28] tidyverse_1.3.0         ngsReports_1.6.1        tibble_3.0.4           
[31] ggplot2_3.3.3           BiocGenerics_0.36.0     workflowr_1.6.2        

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                  backports_1.2.1              
  [3] plyr_1.8.6                    lazyeval_0.2.2               
  [5] crosstalk_1.1.0.1             BiocParallel_1.24.1          
  [7] digest_0.6.27                 htmltools_0.5.0              
  [9] fansi_0.4.1                   memoise_1.1.0                
 [11] cluster_2.1.0                 Biostrings_2.58.0            
 [13] modelr_0.1.8                  matrixStats_0.57.0           
 [15] askpass_1.1                   prettyunits_1.1.1            
 [17] jpeg_0.1-8.1                  colorspace_2.0-0             
 [19] blob_1.2.1                    rvest_0.3.6                  
 [21] rappdirs_0.3.1                ggrepel_0.9.0                
 [23] haven_2.3.1                   xfun_0.20                    
 [25] crayon_1.3.4                  RCurl_1.98-1.2               
 [27] jsonlite_1.7.2                zoo_1.8-8                    
 [29] gtable_0.3.0                  zlibbioc_1.36.0              
 [31] XVector_0.30.0                DelayedArray_0.16.0          
 [33] DBI_1.1.0                     Rcpp_1.0.5                   
 [35] viridisLite_0.3.0             xtable_1.8-4                 
 [37] progress_1.2.2                flashClust_1.01-2            
 [39] bit_4.0.4                     DT_0.17                      
 [41] htmlwidgets_1.5.3             httr_1.4.2                   
 [43] RColorBrewer_1.1-2            ellipsis_0.3.1               
 [45] farver_2.0.3                  pkgconfig_2.0.3              
 [47] XML_3.99-0.5                  here_1.0.1                   
 [49] locfit_1.5-9.4                labeling_0.4.2               
 [51] tidyselect_1.1.0              rlang_0.4.10                 
 [53] reshape2_1.4.4                later_1.1.0.1                
 [55] munsell_0.5.0                 BiocVersion_3.12.0           
 [57] cellranger_1.1.0              tools_4.0.3                  
 [59] cli_2.2.0                     generics_0.1.0               
 [61] RSQLite_2.2.2                 broom_0.7.3                  
 [63] evaluate_0.14                 fastmap_1.0.1                
 [65] ggdendro_0.1.22               knitr_1.30                   
 [67] bit64_4.0.5                   fs_1.5.0                     
 [69] whisker_0.4                   mime_0.9                     
 [71] leaps_3.1                     xml2_1.3.2                   
 [73] biomaRt_2.46.0                compiler_4.0.3               
 [75] rstudioapi_0.13               curl_4.3                     
 [77] png_0.1-7                     interactiveDisplayBase_1.28.0
 [79] reprex_0.3.0                  stringi_1.5.3                
 [81] highr_0.8                     lattice_0.20-41              
 [83] ProtGenerics_1.22.0           Matrix_1.2-18                
 [85] vctrs_0.3.6                   pillar_1.4.7                 
 [87] lifecycle_0.2.0               BiocManager_1.30.10          
 [89] data.table_1.13.6             bitops_1.0-6                 
 [91] httpuv_1.5.4                  rtracklayer_1.50.0           
 [93] R6_2.5.0                      latticeExtra_0.6-29          
 [95] hwriter_1.3.2                 promises_1.1.1               
 [97] ShortRead_1.48.0              gridExtra_2.3                
 [99] MASS_7.3-53                   assertthat_0.2.1             
[101] SummarizedExperiment_1.20.0   openssl_1.4.3                
[103] rprojroot_2.0.2               withr_2.3.0                  
[105] GenomicAlignments_1.26.0      Rsamtools_2.6.0              
[107] GenomeInfoDbData_1.2.4        hms_0.5.3                    
[109] grid_4.0.3                    rmarkdown_2.6                
[111] MatrixGenerics_1.2.0          git2r_0.28.0                 
[113] scatterplot3d_0.3-41          shiny_1.5.0                  
[115] lubridate_1.7.9.2             FactoMineR_2.4               </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
